CRM/ManualDirectDebit/Batch/Transaction.php

```diff
@@ -412,12 +448,20 @@ public function getDDMandateInstructions() {
       }
     }
 
-    if (!empty($this->params['receive_date_start'])) {
-      $query->where('civicrm_contribution.receive_date >= @receive_date_start', ['receive_date_start' => $this->params['receive_date_start']]);
+    // handle date received field.
```

Comment:
maybe it is better to move this to a new private method and give a more clear name,same with "handle cancel date field." below .. and not to be honest this method is getting so big, if I have time to work on it I would probably try to discover the possibility of moving it to its own class.